File: csharp_solutions/valid_anagram.cs
public class Solution {
    public bool IsAnagram(string s, string t) {
        Dictionary<char, int> sCount = new Dictionary<char, int>();
        Dictionary<char, int> tCount = new Dictionary<char, int>();
        if(s.Length != t.Length) {
            return false;
        }
        else {
            for (int i=0; i<s.Length; i++) {
                if (!sCount.ContainsKey(s[i])) {
                    sCount[s[i]] = 1;
                }
                else {
                    sCount[s[i]] ++;
                }
                if (!tCount.ContainsKey(t[i])) {
                    tCount[t[i]] = 1;
                }
                else {
                    tCount[t[i]] ++;
                }
            }
            return this.areDictsEqual(sCount, tCount);
        }
    }
    
    private bool areDictsEqual(Dictionary<char, int> sCount,Dictionary<char, int> tCount) {
        foreach( KeyValuePair<char,int> kvp in sCount ) {
            if (!tCount.ContainsKey(kvp.Key) || tCount[kvp.Key] != kvp.Value) {
                return false;
            }
        }
        return true;
    }
}

File: python_solutions/valid_anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        s_letter_dict = {}
        t_letter_dict = {}
        if len(s) != len(t):
            return False
        for i in range(len(s)):
            s_letter_dict[s[i]] = 1 + s_letter_dict.get(s[i], 0)
            t_letter_dict[t[i]] = 1 + t_letter_dict.get(t[i], 0)
        for key in s_letter_dict:
            if s_letter_dict[key] != t_letter_dict.get(key, 0):
                return False
        return True